pageserver/src/page_service.rs

```diff
@@ -1802,6 +1837,44 @@ where
             // important because psycopg2 executes "SET datestyle TO 'ISO'"
             // on connect
             pgb.write_message_noflush(&BeMessage::CommandComplete(b"SELECT 1"))?;
+        } else if query_string.starts_with("lease lsn ") {
+            let (_, params_raw) = query_string.split_at("lease lsn ".len());
```

Comment:
```suggestion
        } else if let Some(params_raw) = query_string.strip_prefix("lease lsn ") {
```

Reply:
`strip_prefix` is a good idea, but I would prefer to do it in a followup. also present in the other places.

Reply:
Yeah, this would benefit from some separation for parsing and voila, we'd have less duplication as well.